Mathlib/Data/Polynomial/Splits.lean

```diff
@@ -438,6 +438,13 @@ theorem splits_id_of_splits {f : K[X]} (h : Splits i f)
     (roots_mem_range : ∀ a ∈ (f.map i).roots, a ∈ i.range) : Splits (RingHom.id K) f :=
   splits_of_comp (RingHom.id K) i h roots_mem_range
 
+theorem splits_of_algHom {R L' : Type*} [CommRing R] [Field L'] [Algebra R L] [Algebra R L']
```

Comment:
I think this is more naturally stated asking that `L'` is an `L` algebra with `IsScalarTower`. You can surely produce `e` in one line then.

If you really want to keep this version we can also to that, but we want also the other one.

Reply:
Thanks for the review. I added the version with `IsScalarTower` but kept the old version since it is the one I need later on.